lib/credit_card_sanitizer.rb

```diff
@@ -88,14 +88,17 @@ def initialize(options = {})
   #   sanitize!("I want all your credit card numbers!")
   #   #=> nil
   #
-  # Returns a String of the redacted text if a credit card number was detected.
-  # Returns nil if no credit card numbers were detected.
+  # If options[:return_changes] is false, returns nil if no redaction happened,
+  # else the full text after redaction.
+  #
+  # If options[:return_changes] is true, returns nil if no redaction happened,
+  # else an array of [old_text, new_text] indicating what substrings were redacted.
   def sanitize!(text, options = {})
     options = @settings.merge(options)
 
     text.force_encoding(Encoding::UTF_8)
     text.scrub!('�')
-    redacted = nil
+    changes = nil
```

Comment:
might be less indirection to do `changes = []` and then check `changes.any?` at the end

Reply:
I was thinking `changes = []` would be more expensive than `changes = nil` since an object needs to be allocated, and this method gets called on huge streams of incoming text... that's why I do `changes ||= []` only when there's something to actually append to `changes`

Am I over-optimizing?

Reply:
I'd say over-optimizing :D
... write the most readable code 👍

Reply:
... but also can just leave it